src/track/timesignature.h

```diff
@@ -0,0 +1,49 @@
+// signature.h
+// Created on: 06/11/2019 by Javier Vilarroig
+
+#pragma once
+
+namespace mixxx {
+
+// Musical Time signature
+// Right now is to be used for bar detection only, so only the beats per bar is useful
+class TimeSignature final {
+  private:
+    int m_beatsPerVar;
+    int m_noteValue;
+
+  public:
+    inline TimeSignature(int beatsPerBar, int noteValue)
+        : m_beatsPerVar(beatsPerBar),
+          m_noteValue(noteValue) {
+    }
+
+    virtual ~TimeSignature() {}
+
+    inline void setBeats(int beatsPerBar) {
+        m_beatsPerVar = beatsPerBar;
+    }
+
+    inline void setNoteValue(int noteValue) {
+        m_noteValue = noteValue;
+    }
+
+    inline int getBeats() const {
+        return m_beatsPerVar;
+    }
+
+    inline int getNoteValue() const {
+        return m_noteValue;
+    }
+
+    inline bool operator==(const TimeSignature& signature) const {
+        return(signature.m_beatsPerVar == m_beatsPerVar && signature.m_noteValue == m_noteValue);
+    }
+};
+
+// Invalid Signature
+const TimeSignature kNullTimeSignature = TimeSignature(0,0);
```

Comment:
can this become a constexpr?

Reply:
```suggestion
constexpr TimeSignature kNullTimeSignature = TimeSignature(0, 0);
```

Reply:
By the way, are you using the pre-commit hooks? That should make it easy to detect formatting issues before committing them:
https://github.com/mixxxdj/mixxx/blob/master/.pre-commit-config.yaml#L20-L40

Reply:
It fails to compile. Apparently I must include a constexpr constructor and I'm not so sure is worth the effort.